### ISSUE TITLE
Dropdown option values showing in email notifications instead of the dropdown option labels

### ISSUE THREAD

**Issue by Timusan** [verbb/formie#800]
This is basically the same run as #756 , only this occurs with Dropdowns.
The following code was expected to work (as per the example for checkboxes in #756):

```php
        Event::on(Dropdown::class, Dropdown::EVENT_MODIFY_VALUE_FOR_EMAIL, function (ModifyFieldEmailValueEvent $event) {
            // Fetch the raw value again, from the submission (`craft\fields\data\MultiOptionsFieldData`)
            $submissionValue = $event->submission->getFieldValue($event->field->handle);

            // Overwrite the value in emails to use labels instead of values
            $event->value = implode(', ', array_map(function ($item) use ($event) {
                return Craft::t('site', $item->label, [], $event->submission->getFieldValue('lang'));
            }, (array) $submissionValue));
        });
```

Is this maybe not implemented yet for Dropdown fields?

**Comment by engram-design** [comment]
It's a contentious subject on whether we should us values or labels for option fields in email notifications. I've not yet come up with a nice way to configure this (globally is an option, but not ideal), as per https://github.com/verbb/formie/issues/756.

But the event code will work to override things - however with Dropdown fields, values can be a `craft\fields\data\MultiOptionsFieldData` or `craft\fields\data\SingleOptionsFieldData` whereas a Checkboxes field will always be `MultiOptionsFieldData`.

```php
Event::on(Dropdown::class, Dropdown::EVENT_MODIFY_VALUE_FOR_EMAIL, function (ModifyFieldEmailValueEvent $event) {
    // Fetch the raw value again, from the submission
    $submissionValue = $event->submission->getFieldValue($event->field->handle);

    // Overwrite the value in emails to use labels instead of values
    if ($submissionValue instanceof \craft\fields\data\MultiOptionsFieldData) {
        $event->value = implode(', ', array_map(function($item) {
            return $item->label;
        }, (array)$submissionValue));
    } else {
        $event->value = $submissionValue->label ?? '';
    }
});
```

**Comment by Timusan** [comment]
Thanks for the clarification and the updated code @engram-design. However, the main problem seems to be that the event `Dropdown::EVENT_MODIFY_VALUE_FOR_EMAIL` never seems to be triggered.

A breakpoint in `Checkboxes::EVENT_MODIFY_VALUE_FOR_EMAIL` yields a break (and updated values in the email) but a breakpoint in `Dropdown::EVENT_MODIFY_VALUE_FOR_EMAIL` does not stop the code nor are values replaced. 

I have both pieces of code in the same plugin and on the same level:

```php
        Event::on(Checkboxes::class, Checkboxes::EVENT_MODIFY_VALUE_FOR_EMAIL, function (ModifyFieldEmailValueEvent $event) {
            // Fetch the raw value again, from the submission (`craft\fields\data\MultiOptionsFieldData`)
            $submissionValue = $event->submission->getFieldValue($event->field->handle);

            // Overwrite the value in emails to use labels instead of values
            $event->value = implode(', ', array_map(function ($item) use ($event) {
                return Craft::t('site', $item->label, [], $event->submission->getFieldValue('lang'));
            }, (array) $submissionValue));
        });

        Event::on(Dropdown::class, Dropdown::EVENT_MODIFY_VALUE_FOR_EMAIL, function (ModifyFieldEmailValueEvent $event) {
            // Fetch the raw value again, from the submission
            $submissionValue = $event->submission->getFieldValue($event->field->handle);

            // Overwrite the value in emails to use labels instead of values
            if ($submissionValue instanceof \craft\fields\data\MultiOptionsFieldData) {
                $event->value = implode(', ', array_map(function($item) {
                    return $item->label;
                }, (array)$submissionValue));
            } else {
                $event->value = $submissionValue->label ?? '';
            }
        });
```

Can it be that `EVENT_MODIFY_VALUE_FOR_EMAIL` is only implemented for Checkboxes and not yet for Dropdown fields?

**Comment by engram-design** [comment]
Ah, yes that's a good point. The `getValueForEmail()` function (which calls `Field::EVENT_MODIFY_VALUE_FOR_EMAIL`) isn't _quite_ ready for prime-time yet, but it'll be the way going forward once we refactor a few things about email content and value parsing for fields.

As such, yes this event is only called for when the value is a `MultiOptionsFieldData`, but I've just added handling for it to be called for `SingleOptionsFieldData` which covers Dropdown and Radio Button field values.

To get this fix early, change your `verbb/formie` requirement in `composer.json` to:

```
"require": {
  "verbb/formie": "dev-craft-3 as 1.5.11",
  "...": "..."
}
```

Then run `composer update`.

**Comment by engram-design** [comment]
Fixed in [1.5.12](https://github.com/verbb/formie/releases/tag/1.5.12)
